assets/js/base/hooks/use-store-notices.js

```diff
@@ -2,44 +2,49 @@
  * External dependencies
  */
 import { useStoreNoticesContext } from '@woocommerce/base-context';
-import { useMemo } from '@wordpress/element';
 
 export const useStoreNotices = () => {
-	const { notices, createNotice, removeNotice } = useStoreNoticesContext();
+	const {
+		notices,
+		createNotice,
+		removeNotice,
+		createSnackbarNotice,
+	} = useStoreNoticesContext();
 
-	const noticesApi = useMemo(
-		() => ( {
-			addDefaultNotice: ( text, noticeProps = {} ) =>
-				void createNotice( 'default', text, {
-					...noticeProps,
-				} ),
-			addErrorNotice: ( text, noticeProps = {} ) =>
-				void createNotice( 'error', text, {
-					...noticeProps,
-				} ),
-			addWarningNotice: ( text, noticeProps = {} ) =>
-				void createNotice( 'warning', text, {
-					...noticeProps,
-				} ),
-			addInfoNotice: ( text, noticeProps = {} ) =>
-				void createNotice( 'info', text, {
-					...noticeProps,
-				} ),
-			addSuccessNotice: ( text, noticeProps = {} ) =>
-				void createNotice( 'success', text, {
-					...noticeProps,
-				} ),
-			removeNotices: ( type = null ) => {
-				notices.map( ( notice ) => {
-					if ( type === null || notice.status === type ) {
-						removeNotice( notice.id );
-					}
-					return true;
-				} );
-			},
-		} ),
-		[ createNotice ]
-	);
+	const noticesApi = {
+		addDefaultNotice: ( text, noticeProps = {} ) =>
+			void createNotice( 'default', text, {
+				...noticeProps,
+			} ),
+		addErrorNotice: ( text, noticeProps = {} ) =>
+			void createNotice( 'error', text, {
+				...noticeProps,
+			} ),
+		addWarningNotice: ( text, noticeProps = {} ) =>
+			void createNotice( 'warning', text, {
+				...noticeProps,
+			} ),
+		addInfoNotice: ( text, noticeProps = {} ) =>
+			void createNotice( 'info', text, {
+				...noticeProps,
+			} ),
+		addSuccessNotice: ( text, noticeProps = {} ) =>
+			void createNotice( 'success', text, {
```

Comment:
Related to above (single API for all notices) – I'm wondering if we can smart-default the UI based on severity here, to encourage consistent use. So `addErrorNotice` would default to notice, `addSuccessNotice` would default to snackbar, etc. This could still potentially be overridden if needed, by passing `noticeProps = { type: 'snackbar' }` or `noticeProps = { type: 'notice' }`.

This interface is probably stretching the meaning of the term "notice" though.

Not a blocker, just exploring this as a potential improvement to the API we expose to clients.

Reply:
Currently I'm following the pattern established by Gutenberg for the [usage of Snackbars](https://github.com/WordPress/gutenberg/tree/master/packages/components/src/snackbar) which doesn't involve varying degrees of styles or behaviour.

So with that in mind,  it seems pretty clear to me that snackbar notices have an explicit purpose that differs from that of regular notices.  Whether that means all success and/or info notices will always be snackbar notices is not so clear to me so I think it's best to keep them distinct for now.